### DISCUSSION TITLE
[Basic elisp question] How can I define a function that calls `meow-end-of-thing` -> "l"?

### DISCUSSION THREAD

**Discussion by kartikynwa** [meow-edit/meow#643] [Q&A] [answered]
I am trying to define a function that is equivalent of (meow-end-of-thing) then pressing the "l" key.

From meow-end-of-thing's definition I see that it takes a `thing` argument. So I assumed that if I call this function non-interactively I can supply what the thing should be. But I haven't been able to get it to work. I have tried this

```elisp
(defun meow-end-of-line ()
  (interactive)
  (meow-end-of-thing 'line))
```

But I haven't been able to wrap my head around symbols in elisp yet. Any help is welcome. Thanks.

**Comment by DogLooksGood** [comment, marked as answer]
Hey, the argument THING accepts a char, not a symbol. 

An inner function `meow--parse-range-of-thing` gives you the bounds by a thing symbol. Or you can do this:

```
(defun meow-end-of-line ()
  (interactive)
  (meow--select (meow--make-selection 'line (point) (line-end-position))))
```

**Reply by kartikynwa** [reply]
Thanks a lot. `(meow-end-of-thing ?l)` worked perfectly. In hindsight I should have been able to guess the expected type of the argument seeing how it takes input from a transient menu in interactive mode. But I tunnel visioned on trying to understand symbols in elisp. Thanks very mucb again.
